app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/newdevicenotice/NewDeviceNoticeTwoFactorScreenTest.kt

```diff
@@ -0,0 +1,108 @@
+package com.x8bit.bitwarden.ui.auth.feature.newdevicenotice
+
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.core.net.toUri
+import com.x8bit.bitwarden.data.platform.repository.util.bufferedMutableSharedFlow
+import com.x8bit.bitwarden.ui.platform.base.BaseComposeTest
+import com.x8bit.bitwarden.ui.platform.manager.intent.IntentManager
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.verify
+import junit.framework.TestCase.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class NewDeviceNoticeTwoFactorScreenTest : BaseComposeTest() {
+    private val intentManager = mockk<IntentManager>(relaxed = true) {
+        every { startCustomTabsActivity(any()) } just runs
+    }
+    private var onNavigateBackCalled = false
+    private val mutableEventFlow = bufferedMutableSharedFlow<NewDeviceNoticeTwoFactorEvent>()
+    private val viewModel = mockk<NewDeviceNoticeTwoFactorViewModel>(relaxed = true) {
+        every { eventFlow } returns mutableEventFlow
+    }
+
+    @Before
+    fun setUp() {
+        composeTestRule.setContent {
+            NewDeviceNoticeTwoFactorScreen(
+                onNavigateBack = { onNavigateBackCalled = true },
```

Comment:
You may want to set `onNavigateBackCalled` to false in a `@After tearDown()` function. Otherwise it could result in false positives if test B checks the value after it's set to true on test A, but is never actually called in test B.